packages/commerce-sdk-react/src/hooks/types.ts

```diff
@@ -81,13 +45,43 @@ export interface QueryParams {
     shortCode?: string
 }
 
-export interface QueryResponse<T> extends CommonHookResponse {
-    data: T
+export type QueryResponse<T> =
+    | {
+          isLoading: boolean
+          error?: undefined
+          data?: undefined
+      }
+    | {
+          isLoading: true
+          error?: Error
+          data?: T
+      }
+    | {
+          isLoading: false
+          error: Error
+          data?: undefined
+      }
+    | {
+          isLoading: false
+          error?: undefined
+          data: T
+      }
+
+// TODO: Include action name as a key in addition to "execute"?
+export type ActionResponse<A, R> = QueryResponse<R> & {
+    execute: (arg: A) => void
```

Comment:
It would be nice if we can replace `execute` with the actual action name. But I'm not sure how to achieve that with my poor typescript skill...

Reply:
I tried, but couldn't quite crack it, and decided to leave it for later.

Reply:
If we replace `execute` with the action name:
- How about `isLoading` and `error`? They still need to be renamed.
- What if users do not do destructuring? They would see `addToBasket.addToBasket()` (instead of `addToBasket.execute()`).

IMO it's better to keep using `execute`.

Reply:
I was planning on having both available, so that users can choose whichever they prefer.